server/app/com/xsn/explorer/services/TransactionCollectorService.scala

```diff
@@ -217,6 +223,58 @@ class TransactionCollectorService @Inject()(
       case Bad(e) => Bad(e)
     }
   }
+
+  private[services] def getTposContracts(
+      transactions: List[persisted.Transaction.HasIO]
+  ): FutureListOps[TPoSContract] = {
+
+    val futures = transactions.map { transaction =>
+      val result = for {
+        contractDetails <- xsnService.getTPoSContractDetails(transaction.id).toFutureOr
+
+      } yield {
+        val id = TPoSContract.Id(transaction.id, 1)
```

Comment:
Why do you think it must be a 1? I think it should be the output index where the collateral is found, please verify with the previous code.

Reply:
fixed